### PR TITLE
Fix breaking changes introduced in the NCBI API

### PR DESCRIPTION
**Pull request by drpatelh**
See https://github.com/nf-core/fetchngs/issues/99

NCBI recently updated their base API URLs which broke the functionality in this pipeline to fetch metadata for SRA and GEO ids.

I can't find a way to obtain GEO metadata using the current APIs so I will strip this functionality out of the pipeline and add it back in when a robust solution exists.

## ✅ Fixed for SRA ids

I have fixed the base URL for fetching SRA metadata based on the docs [here](https://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EFetch)

### Old URL

`https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?save=efetch&db=sra&rettype=runinfo&term=SRR14593545`

### New URL

`https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?id=SRR14593545&db=sra&rettype=runinfo&retmode=text`

##  :x: Broken for GEO ids

I tried to update the base URL for fetching GEO metadata based on the docs [here](https://www.ncbi.nlm.nih.gov/geo/info/geo_paccess.html#Examples) but it is still broken....

### Old URL

`https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM4907283&targ=gsm&view=data&form=text`

### New URL

`https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?id=GSM4907283&db=gds&rettype=runinfo&retmode=text`

### CODE REVIEW

File: assets/schema_input.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
         "items": {
             "type": "string",
             "pattern": "^(((SR|ER|DR)[APRSX])|(SAM(N|EA|D))|(PRJ(NA|EB|DB))|(GS[EM])|(syn))(\\d+)$",
```
Comment (maxulysse) [inline assets/schema_input.json:11]: you kept GS[EM] in the pattern

**Review by maxulysse — APPROVED** [review]

**Comment by hukai916** [comment]
To fetch metadata given GEO id, will ffq help: https://github.com/pachterlab/ffq ? Seems that ffq calls esearch utilities first and parse the results to obtain a full meta table.

**Comment by drpatelh** [comment]
Hi @hukai916 ! Yep, I have already created a draft PR to add in a `ffq` implementation https://github.com/nf-core/fetchngs/pull/100 and just created an issue to officially track this feature https://github.com/nf-core/fetchngs/issues/110

**Comment by Midnighter** [comment]
Using esearch first also works here:

1. https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=sra&term=GSM4907283
2. https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=sra&id=12420761&rettype=runinfo&retmode=text

**Comment by ejseqera** [comment]
An eutils workflow for working with GEO IDs is somewhat described here: https://www.ncbi.nlm.nih.gov/geo/info/geo_paccess.html (updated March 27 2023)

**Pull request merged**